designs/dedicatedhosts.md

```diff
@@ -0,0 +1,86 @@
+
+## Solutions
+
+aws-karpenter already supports a set of [LaunchTemplate configuration](https://github.com/aws/karpenter/blob/main/pkg/apis/v1alpha1/awsnodetemplate.go)
+with design document: https://github.com/aws/karpenter/blob/main/designs/aws-launch-templates-v2.md
+
+1. Implement the minimal fields from https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata-placement.html
+specifically `placement.hostResourceGroupArn` and `licenseConfiguration`
+
+this would look like:
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  licenseConfiguration: arn:aws:license-manager:eu-east-1:123456789012:license-configuration:lic-edf7f9e241f5e16f29996c842111f448 # optional, arn of the license configuration
+  placement:
+    hostResourceGroupArn: arn:aws:resource-groups:us-east-1:123456789012:group/my-hrg-name #optional, The ARN of the host resource group in which to launch the instances. If you specify a host resource group ARN, omit the Tenancy parameter or set it to host.
+  
+```
+
+2. Implement the complete fields from AWS Launch Templates
+
+eg.
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  licenseConfiguration: arn:aws:license-manager:eu-east-1:123456789012:license-configuration:lic-edf7f9e241f5e16f29996c842111f448 # optional, arn of the license configuration
+  placement:
+    affinity: #optional
+    availabilityZone: #optional
+    groupId: #optional, something to do with placement groups
+    groupName: #optional
+    hostId: #optional, ID of the dedicated host
+    hostResourceGroupArn: arn:aws:resource-groups:us-east-1:123456789012:group/my-hrg-name #optional, The ARN of the host resource group in which to launch the instances. If you specify a host resource group ARN, omit the Tenancy parameter or set it to host.
+    paritionNumber: #optional, The number of the partition the instance should launch in. Valid only if the placement group strategy is set to partition.
+    spreadDomain: #optional, reserved for future use
+    tenancy: dedicated #optional, The tenancy of the instance. An instance with a tenancy of dedicated runs on single-tenant hardware, one of dedicated | default | host
+
+```
+
+3. Implement a simplified API
+AWS Launch templates also include extra fields
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  dedicatedHostConfig:
+    licenseConfiguration: arn:aws:license-manager:eu-east-1:123456789012:license-configuration:lic-edf7f9e241f5e16f29996c842111f448 # optional, arn of the license configuration
+    resourceHostGroup: arn:aws:resource-groups:us-east-1:123456789012:group/my-hrg-name #option, arn of the HRG 
+```
+
+## Recommendations
+
+1. Middle ground solution, stick both to the AWS Launch Template api, but implementing the smallest set of configuration practical, 
```

Comment:
So you're suggesting Option 2 but only include `licenseConfiguration` and `placement.hostResourceGroupARN`? 

Are `placement.tenancy` or `placement.hostId` also needed? 

Overall, I agree with that approach of keeping to the LT API but only implement what we need right now.

Reply:
Correct. Afaik `placement.tenancy` is assumed by AWS to be dedicated when a `hostResourceGroupArn` is provided. 

I think this is the only scenario karpenter should consider as it treats a group of dedicated hosts as a single pool of resources. 

We should favour `hostResourceGroupArn` over `hostId` as we dont want to be making fine grained scheduling decisions like which ec2 instancr goes on which DH. Also HRGs can exist with a single DH. 

It is possible to to set `placement.tenancy=dedicated` and not provide a hrg iirc. But this seems like a crazy way to spend money, as it allows AWS to provision dedicated hosts for you (which can be crazy expensive).

Reply:
Option 2 sounds good for me. 👍